common/src/main/java/tachyon/security/UserInformation.java

```diff
@@ -15,109 +15,112 @@
 
 package tachyon.security;
 
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
+import javax.security.auth.Subject;
 import javax.security.auth.login.AppConfigurationEntry;
 import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import com.google.common.annotations.VisibleForTesting;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.security.authentication.AuthenticationFactory;
 import tachyon.util.PlatformUtils;
 
 /**
- * This class wraps a JAAS Subject and provides methods to determine the user. It supports
- * Windows, Unix, and Kerberos login modules.
+ * This class provides methods to determine the login user and connected remote client users.
+ * When fetching a login user of Tachyon service or Tachyon client, it supports Windows, Unix,
+ * and Kerberos login modules.
+ * (TODO) When creating a client user, it instantiates a {@link tachyon.security.User} by the user
+ * name transmitted by SASL transport from client side.
  */
 public class UserInformation {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  private static final String OS_LOGIN_MODULE_NAME;
-  private static final String OS_PRINCIPAL_CLASS_NAME;
-  private static final boolean WINDOWS = PlatformUtils.OS_NAME.startsWith("Windows");
-  private static final boolean IS_64_BIT = PlatformUtils.PROCESSOR_BIT.contains("64");
-  private static final boolean AIX = PlatformUtils.OS_NAME.equals("AIX");
+  /** The configuration to use */
+  private static final TachyonConf TACHYON_CONF;
+  /** The authentication method to use. */
+  private static AuthenticationFactory.AuthType sAuthType;
+  /** User instance of the login user in Tachyon process (Tachyon service or Tachyon client) */
+  private static User sLoginUser;
```

Comment:
If the user is not changed in the lifespan of this instance. Can we make it private final?